lib/utils.js

```diff
@@ -1,8 +1,48 @@
 'use strict';
 
-var fs = require('fs');
-var path = require('path');
-var child_process = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+const https = require('https');
+const child_process = require('child_process');
+const KiteError = require('./kite-error');
+
+function promisifyRequest(request) {
+  return new Promise((resolve, reject) => {
+    request.on('response', resp => resolve(resp));
+    request.on('error', err => reject(err));
+  });
+}
+
+function promisifyStream(stream) {
+  return new Promise((resolve, reject) => {
+    stream.on('finish', () => resolve());
+    stream.on('error', err => reject(err));
+  });
+}
+
+function request(url) {
+  return url.indexOf('https://') === 0
+    ? https.request(url)
+    : http.request(url);
+}
+
+// Given a request this function will follow the redirection until a
+// code different that 303 is returned
+function followRedirections(req) {
+  return promisifyRequest(req)
+  .then(resp => {
+    if (resp.statusCode === 200) {
+      return resp;
+    } else if (resp.statusCode === 303) {
```

Comment:
True, when testing with a dropbox link I got a 302, I'll take a look at that available APIs.